benchmarks/benchmarks/signal_filtering.py

```diff
@@ -82,3 +84,54 @@ def setup(self, n_samples, order):
 
     def time_sosfilt_basic(self, n_samples, order):
         sosfilt(self.sos, self.y)
+
+
+class MedFilt2D(Benchmark):
+    def setup_cache(self):
+        np.random.seed(8176)
+        data = np.random.randn(2000, 4000)
+        expected = medfilt2d(data, 5)
+        np.save("medfilt2d_data.npy", data)
+        np.save("medfilt2d_expected.npy", expected)
+
+    def setup(self):
+        self.data = np.load("medfilt2d_data.npy")
+        self.expected = np.load("medfilt2d_expected.npy")
+
+    def teardown(self):
+        assert_allclose(self.output, self.expected)
+
+    def _multithreaded(self):
+        # Lets take four overlapping chunks over the last axis. First pair of
+        # values is range to select from input, second pair range of the result
+        # to use, last pair the range of the output array to store it in.
+        chunks = (
+            (0, 1002, None, -2, 0, 1000),
+            (998, 2002, 2, -2, 1000, 2000),
+            (1998, 3002, 2, -2, 2000, 3000),
+            (2998, 4000, 2, None, 3000, 4000)
+        )
+
+        self.output = np.empty(self.data.shape, dtype=self.data.dtype)
+
+        def apply(chunk):
+            return medfilt2d(self.data[:, chunk[0]:chunk[1]], 5)
+
+        with ThreadPoolExecutor(max_workers=4) as executor:
+            futures = {executor.submit(apply, chk): chk for chk in chunks}
+            for future in as_completed(futures):
+                chk = futures[future]
+                result = future.result()
+                self.output[:, chk[4]:chk[5]] = result[:, chk[2]:chk[3]]
+
+    def time_singlethreaded(self):
+        self.output = medfilt2d(self.data, 5)
+
+    def time_multithreaded(self):
+        self._multithreaded()
```

Comment:
The output of this is a single number:
```
[ 37.50%] ··· Setting up signal_filtering.py:90                                 ok
[ 37.50%] ··· Running (signal_filtering.MedFilt2D.time_multithreaded--)..
[ 62.50%] ··· signal_filtering.MedFilt2D.peakmem_multithreaded                311M
[ 75.00%] ··· signal_filtering.MedFilt2D.peakmem_singlethreaded               247M
[ 87.50%] ··· signal_filtering.MedFilt2D.time_multithreaded             2.51±0.05s
[100.00%] ··· signal_filtering.MedFilt2D.time_singlethreaded            2.37±0.01s
```

Compare with `ParallelSosfilt`:
```
[100.00%] ··· signal_filtering.ParallelSosfilt.time_sosfilt                     ok
[100.00%] ··· =========== ============ ============ ============
              --                         threads                
              ----------- --------------------------------------
               n_samples       1            2            4      
              =========== ============ ============ ============
                 1000.0    77.8±0.6ms   41.3±0.2ms   22.7±0.2ms 
                10000.0     310±30ms     230±30ms     194±3ms   
              =========== ============ ============ ============
```

If you can change this to follow the `ParallelSosfilt` structure (can do one shape or a couple of different shapes if the performance varies), that would be great.